highfidelity/hifi
# avatar mixer bandwidth optimization

Changes in this PR:
- changes to the packing format of the AvatarData. now each section is only included if it changes.
- use 8bits for audio loudness
- local position is only included if there is a parent
- small changes in joint data is now culled in the mixer.  The tolerance increases (larger changes required to send) at greater distances.

Test Plan:
- Needs a smoke test with *multiple* avatars connecting to mixer and verify that "things still work"
- Do you see animations and hand movements?
- Do you see mouth animations based on audio loudness?
- Make sure to test "avatar riding an entity" (something setting the parent of the avatar to an entity).
- Make sure to test independent camera mode and 3rd person.
- Make sure to test avatar resizing.
- Make sure to test space bubble.
- Make sure to test that out of view avatars send less data.
- Run the debugAvatarMixer.js script, and see that the "parts" of the avatar don't send every frame.
- If you get really really far away from an avatar, even if it would be in your view, you should see the bandwidth drop pretty significantly. (Note: really far away is defined as 10km away, in which case you won't be able to see the avatar or the overlay, but you can tell by your total avatar mixer bandwidth that things have dropped).

Note: 
- There is an existing bug (in master, beta) where ~10 seconds the avatar mixer seems to send less joint data. That problem can be seen in master, and is not fixed by this PR.
